src/core/i18n/utils.js

```diff
@@ -278,19 +278,29 @@ export function makeI18n(
     i18n.options._momentDefineLocale();
   }
 
-  // This makes sure moment always uses the current locale.
-  moment.locale(makeMomentLocale(i18n.lang));
-
-  // Wrap the core Jed functionality so that we can always strip leading whitespace
-  // from translation keys to match the same process used in extraction.
+  // Wrap the core Jed functionality so that we can always strip leading
+  // whitespace from translation keys to match the same process used in
+  // extraction.
   i18n._dcnpgettext = i18n.dcnpgettext;
   i18n.dcnpgettext = function dcnpgettext(domain, context, singularKey, pluralKey, val) {
     return i18n._dcnpgettext(domain, context, oneLineTranslationString(singularKey),
       oneLineTranslationString(pluralKey), val);
   };
 
-  // We add a translated "moment" property to our `i18n` object
-  // to make translated date/time/etc. easy.
-  i18n.moment = moment;
+  const momentLocale = makeMomentLocale(i18n.lang);
+
+  // We add a translated "moment" property to our `i18n` object to make
+  // translated date/time/etc. easy.
+  i18n.moment = (...params) => {
+    const scopedMoment = moment(...params);
+
+    // This also makes sure moment always uses the current locale.
+    scopedMoment.locale(momentLocale);
```

Comment:
I'm slightly nervous about calling `.locale()` every single time we need to format a string. I took a very quick look and it seems to [use caching](https://github.com/moment/moment/blob/6beb383a0ab0d4c8b3bb56d1393d810db79d486e/src/lib/locale/locales.js#L60) but I didn't inspect everything.

What about this? As far as I can tell [set()](https://momentjs.com/docs/#/get-set/set/) accomplishes the same thing.

```diff
diff --git a/src/core/i18n/utils.js b/src/core/i18n/utils.js
index 5dabcb426..521151508 100644
--- a/src/core/i18n/utils.js
+++ b/src/core/i18n/utils.js
@@ -288,16 +288,12 @@ export function makeI18n(
   };
 
   const momentLocale = makeMomentLocale(i18n.lang);
+  const scopedMoment = moment().locale(momentLocale);
 
   // We add a translated "moment" property to our `i18n` object to make
   // translated date/time/etc. easy.
   i18n.moment = (...params) => {
-    const scopedMoment = moment(...params);
-
-    // This also makes sure moment always uses the current locale.
-    scopedMoment.locale(momentLocale);
-
-    return scopedMoment;
+    return scopedMoment.set(...params);
   };
 
   i18n.moment.locale = () => momentLocale;

```

Reply:
Are you sure about this `set()` method? From the doc, it is a...:

> Generic setter, accepting unit as first argument, and value as second

Reply:
I tried to find a constructor that is not `moment()` in the doc but could not find it 😢 I have seen various issues mentioning this approach of calling `locale()` right before `format()`, hence my patch. Of course, I understand your concern.

Reply:
I just tried and I can confirm it does not work. Tests pass though because.. well, we dont have a test for that I'd say. Going to add one.

Reply:
Test case added here: https://github.com/mozilla/addons-frontend/pull/3844/commits/58245a8b0a9498976f575afde703854b76785897.

Reply:
Oh wow, good catch! Thanks for the test. I find `moment` so utterly confusing.